src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonNodeTests.cs

```diff
@@ -39,6 +41,10 @@ public static void JsonTypes_Deserialize()
             Assert.IsAssignableFrom<JsonValue>(JsonNode.Parse("\"str\""));
             Assert.IsAssignableFrom<JsonValue>(JsonNode.Parse(ToUtf8("\"str\"")));
             Assert.IsType<JsonElement>(JsonSerializer.Deserialize<object>("\"str\""));
+
+            Assert.Null(JsonSerializer.Deserialize<JsonNode>("null"));
```

Comment:
Would it be possible to 

1. Also test deserializing an object containing a `JsonNode` property?
2. Add tests validating the same behavior for `JsonObject` and `JsonArray`?

Reply:
Sure thing!